### DISCUSSION TITLE
Adding existing PDF inside my generated PDF

### DISCUSSION THREAD

**Discussion by Placeprom** [diegomura/react-pdf#2194] [Q&A]
Hey there, 

I am using react-pdf/renderer to generate pdfs with some Images as attachements which I fetch from my server. This works flawlessly. I would like to add previously uploaded pdfs aswell but did not have any luck with it. Is there anything i missed in the documentation? My take would be to convert the pdfs to images first which seems way too hacky. 

Help is much appreciated!

Cheers!

**Comment by diegomura** [comment]
Hi! No, this is not supported. react-pdf only generates new PDF documents. It cannot parse an existing PDF file, so there is no way to add uploaded PDFs directly into your generated document.

Two common workarounds:

1. Convert the pages of the uploaded PDFs to images and place them with the `Image` component.
2. Generate your document with react-pdf first, then merge it with the uploaded PDFs using a library like [pdf-lib](https://github.com/Hopding/pdf-lib) as a second step.

Closing this one since embedding existing PDFs is out of scope for this library. Thanks!
